File: src/main/java/icu/takeneko/omms/client/data/system/DirectoryInfo.java
package icu.takeneko.omms.client.data.system;

import icu.takeneko.omms.client.util.Result;
import lombok.Getter;

import java.util.List;

@Getter
public class DirectoryInfo {
    List<String> folders = null;
    List<String> files = null;

    private final Result result = Result.UNDEFINED;

    public DirectoryInfo() {
    }

    public DirectoryInfo(List<String> folders, List<String> files) {
        this.folders = folders;
        this.files = files;
    }

}

File: src/main/java/icu/takeneko/omms/client/data/system/DirectoryInfo.kt
package icu.takeneko.omms.client.data.system

import icu.takeneko.omms.client.util.Result

@Suppress("Unused")
data class DirectoryInfo(
    var folders: List<String>,
    var files: List<String>,
    var result: Result = Result.UNDEFINED
)
